core/peerset/src/lib.rs

```diff
@@ -103,201 +172,174 @@ pub struct PeersetConfig {
 ///
 /// Implements the `Stream` trait and can be polled for messages. The `Stream` never ends and never
 /// errors.
-pub struct PeersetMut {
-	parent: Arc<Peerset>,
-	rx: mpsc::UnboundedReceiver<Message>,
+#[derive(Debug)]
+pub struct Peerset {
+	data: PeersetData,
+	handle: PeersetHandle,
+	rx: mpsc::UnboundedReceiver<Action>,
+	message_queue: VecDeque<Message>,
+}
+
+impl ops::Deref for Peerset {
```

Comment:
I don't think it's a good use case for `Deref`, from the docs:
> Deref should only be implemented for smart pointers to avoid confusion.

Reply:
hmm... I was not aware of this. I added `Deref` for convenience

Reply:
should I remove this `ops::Deref`?

Reply:
Yeah, if it's ok to use it directly. Alternatively add `AsRef` if you want to pass it to some generic code.